GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -490,94 +455,21 @@ public void DisableContextMenu()
-        private void FormatQuickFilter(RevisionFilter revisionFilter,
-                                       out string revListArgs,
-                                       out string inMemMessageFilter,
-                                       out string inMemCommitterFilter,
-                                       out string inMemAuthorFilter)
-        {
-            revListArgs = string.Empty;
-            inMemMessageFilter = string.Empty;
-            inMemCommitterFilter = string.Empty;
-            inMemAuthorFilter = string.Empty;
-
-            if (!string.IsNullOrEmpty(revisionFilter.Text))
-            {
-                // hash filtering only possible in memory
-                bool cmdLineSafe = GitVersion.Current.IsRegExStringCmdPassable(revisionFilter.Text);
-                revListArgs = " --regexp-ignore-case ";
-                if (revisionFilter.FilterByCommit)
-                {
-                    if (cmdLineSafe && !ObjectId.IsValidPartial(revisionFilter.Text, minLength: 5))
-                    {
-                        revListArgs += "--grep=\"" + revisionFilter.Text + "\" ";
-                    }
-                    else
-                    {
-                        inMemMessageFilter = revisionFilter.Text;
-                    }
-                }
-
-                if (revisionFilter.FilterByCommitter && !string.IsNullOrWhiteSpace(revisionFilter.Text))
-                {
-                    if (cmdLineSafe)
-                    {
-                        revListArgs += "--committer=\"" + revisionFilter.Text + "\" ";
-                    }
-                    else
-                    {
-                        inMemCommitterFilter = revisionFilter.Text;
-                    }
-                }
-
-                if (revisionFilter.FilterByAuthor && !string.IsNullOrWhiteSpace(revisionFilter.Text))
-                {
-                    if (cmdLineSafe)
-                    {
-                        revListArgs += "--author=\"" + revisionFilter.Text + "\" ";
-                    }
-                    else
-                    {
-                        inMemAuthorFilter = revisionFilter.Text;
-                    }
-                }
-
-                if (revisionFilter.FilterByDiffContent)
-                {
-                    if (cmdLineSafe)
-                    {
-                        revListArgs += "-G" + revisionFilter.Text.Quote();
-                    }
-                    else
-                    {
-                        throw new InvalidOperationException(string.Format(_invalidDiffContainsFilter.Text, revisionFilter.Text));
-                    }
-                }
-            }
-        }
-
-        public void SetAndApplyBranchFilter(string filter, bool requireRefresh)
+        public void SetAndApplyBranchFilter(string filter)
         {
-            AppSettings.BranchFilterEnabled = !string.IsNullOrWhiteSpace(filter);
+            // TODO: clean up and move all internals to FilterInfo
```

Comment:
Will this and similar TODO be added to this PR?

Reply:
Some of them - yes, some of them I'll likely leave for another time.